### DISCUSSION TITLE
SAT-Solver-based Gate Schedule Configuration - Showcase

### DISCUSSION THREAD

**Discussion by adamgeorge309** [inet-framework/inet#792] [Showcases]
Discuss the "SAT-Solver-based Gate Schedule Configuration" showcase here. Ideas and recommendations are welcome.

**Comment by tuantitu22** [comment]
Hello, is there any guidance about how to achieve the prerequisition “Z3 Gate Scheduling Configurator INET feature to be enabled, and the libz3-dev or z3-devel packages to be installed” ? Thank you very much!

**Comment by avarga** [comment]
About z3:
On Ubuntu, or any Debian-derivative: `sudo apt install libz3-dev`
On Windows, it's already part of the OMNeT++ bundle.

To enable the INET project feature:
Select the "inet" project, then choose Project -> Project Features from menu, then check the "Z3GateSchedulingConfigurator" item in the list and click Apply.

**Reply by tuantitu22** [reply]
It works now. Thank you very much!

**Comment by jafficy** [comment]
Hello, I have the following problem when running the SatShowcase.anf file, as follows, do I need to install the dependency of optimparallel, looking forward to your reply，Thank you very much!
![2c68be397a7f98d7aacad0c9e7a07dc](https://github.com/inet-framework/inet/assets/115002212/2a07ad90-6ace-49c1-837c-e3d237745faf)
